### ISSUE TITLE
Bad code generation after view_as on enum-struct array argument

### ISSUE THREAD

**Issue by peace-maker** [alliedmodders/sourcepawn#126]
Passing a reference to an array in an enum-struct is broken when the parameter requires a different tag than the array has in the enum-struct.
Tested in SourcePawn Compiler 1.9.0.6115.

Here's how I noticed it - the data written to the file was garbage after converting the plugin to the transitional syntax.
```sourcepawn
enum Test {
	Float:vec3[3],	
	Float:vec2[2]
};

public void main() {
	int test[Test];
	File hFile = OpenFile(..);
	WriteFile(hFile, test[vec3], 3, 4); // warning: tag mismatch but works
	WriteFile(hFile, _:test[vec3], 3, 4); // All fine
	WriteFile(hFile, view_as<int>(test[vec3]), 3, 4); // Writes random memory
}
```

[testcase.zip](https://github.com/alliedmodders/sourcepawn/files/1112612/testcase.zip)
[testcase_shell.zip](https://github.com/alliedmodders/sourcepawn/files/1112756/testcase_shell.zip)

The attached testcase is a shorter version without the need of sourcemod natives.

The assembly of `Bar` and `Bar2` is equivalent:
```
0xd90: break                           
0xd94: break                           
0xd98: push.c 0x2                       ; 2
0xda0: load.s.pri 0xc                   ; test
0xda8: add.c 0xc                        ; 12
0xdb0: push.pri                        
0xdb4: push.c 0x2                       ; 2
0xdbc: call 0xe50                       ; Foo
0xdc4: zero.pri                        
0xdc8: retn
```

`Bar3` uses `view_as<>()` and generates an extra `load.i` in between:
```
0xe10: break                           
0xe14: break                           
0xe18: push.c 0x2                       ; 2
0xe20: load.s.pri 0xc                   ; test
0xe28: add.c 0xc                        ; 12
0xe30: load.i                          
0xe34: push.pri                        
0xe38: push.c 0x2                       ; 2
0xe40: call 0xe50                       ; Foo
0xe48: zero.pri                        
0xe4c: retn
```

**Comment by Fyren** [comment]
I don't think this is limited to view_as: https://pastebin.com/wX0sMehP
Originally from https://forums.alliedmods.net/showthread.php?p=2444797 but I guess I forgot about it until now.

**Comment by peace-maker** [comment]
You are right. "enum-storage" seems only partially supported anyways. Array sizes aren't compared correctly too.
```sourcepawn
enum Test
{
	Float:foo[3]
};

public void OnPluginStart()
{
	int test[Test];
	bar(test[foo]); // error 047: array sizes do not match, or destination array is too small
}

void bar(float rust[3])
{
	PrintToServer("%f", rust[0]);
}
```

**Comment by dvander** [comment]
The view_as issue should be fixed by #129.
